samples/bluetooth/nrf_dm/src/peer.c

```diff
@@ -127,19 +127,43 @@ static uint16_t meter_to_decimeter(float value)
 
 static void ble_notification(const struct peer_entry *peer)
 {
+	struct bt_ddfs_distance_measurement measurement;
+
 	if (!peer) {
 		return;
 	}
 
-	float res;
+	if (peer->result.quality == DM_QUALITY_OK) {
+		measurement.quality = DDFS_QUALITY_OK;
+	} else if (peer->result.quality == DM_QUALITY_POOR) {
+		measurement.quality = DDFS_QUALITY_POOR;
+	} else if (peer->result.quality == DM_QUALITY_DO_NOT_USE) {
+		measurement.quality = DDFS_QUALITY_DO_NOT_USE;
+	} else if (peer->result.quality == DM_QUALITY_CRC_FAIL) {
+		measurement.quality = DDFS_QUALITY_DO_NOT_USE;
+	} else {
+		measurement.quality = DDFS_QUALITY_NONE;
+	}
 
 	if (peer->result.ranging_mode == DM_RANGING_MODE_RTT) {
-		res = peer->result.dist_estimates.rtt.rtt;
+		measurement.ranging_mode = DDFS_DM_RANGING_MODE_RTT;
+		measurement.dist_estimates.rtt.rtt =
+				meter_to_decimeter(peer->result.dist_estimates.rtt.rtt);
 	} else {
-		res = peer->result.dist_estimates.mcpd.best;
+		measurement.ranging_mode = DDFS_DM_RANGING_MODE_MCPD;
+		measurement.dist_estimates.mcpd.ifft =
+				meter_to_decimeter(peer->result.dist_estimates.mcpd.ifft);
+		measurement.dist_estimates.mcpd.phase_slope =
+				meter_to_decimeter(peer->result.dist_estimates.mcpd.phase_slope);
+		measurement.dist_estimates.mcpd.rssi_openspace =
+				meter_to_decimeter(peer->result.dist_estimates.mcpd.rssi_openspace);
+		measurement.dist_estimates.mcpd.best =
+				meter_to_decimeter(peer->result.dist_estimates.mcpd.best);
 	}
 
-	bt_hrs_notify(meter_to_decimeter(res));
+	bt_addr_le_copy(&measurement.bt_addr, &peer->bt_addr);
+
+	bt_ddfs_distance_measurement_notify(NULL, &measurement);
```

Comment:
display error information

Reply:
General question: is it possible to move the whole DDFS handling logic into a separate source file? It currently scattered around two files: main.c and peer.c

Reply:
I moved the logic related to the DDFS to the service.c file